tutorials/generating-synthetic-data/synthetic-data-generation-6-draft.md

```diff
@@ -0,0 +1,225 @@
+---
+myst:
+  html_meta:
+    keywords: "Inductiva API, Programming, HPC, Simulation"
+orphan: true
+---
+
+# Benchmark Computational Resources
+
+In the previous step of this tutorial, we used Inductiva’s templating mechanism to 
+transform the configuration files for our base case simulation into a generalized 
+file whose parameters we could programmatically change. This way, we can systematically 
+produce many variations of the base case we are interested in so that we can then 
+simulate to generate a diverse enough dataset for training our ML models. The 
+parameters that we transformed into variables were the dimensions, initial position and 
+initial velocity of the fluid block, as well as the density and viscosity of the fluid 
+itself. These parameters control the core physical aspects of the simulation, and we 
+will need to cover a wide range of values over these parameters to create a rich 
+dataset. 
+
+Additionally, we also generalized a few hyperparameters that control how the simulation is performed and the level of fidelity, namely: ***particle radius***.
+
+- How much simulation time do we want to generate (e.g. 5 seconds of fluid splashing)?
+- How small are the particles that we use to represent the fluid?        
+
+These hyperparameters are not related to the physical aspects of the simulation and
+only affect the underlying algorithm to improve the numerical stability and resolution.
```

Comment:
@HenriquePreto try to integrate these into our intro and ill do a round of edits after! *almost there*

Reply:
"Additionally, we generalized certain hyperparameters that influence the simulation’s fidelity and performance.
For example, we adjusted the ***period of time*** and the ***particle radius*** to control both the computational effort and the execution time of the simulation.

The first hyperparameter, the ***period of time***, determines how long we want the simulation to run. For example, do we want 5 seconds or 10 seconds of fluid splashing? Naturally, the longer the ***period of time***, the more computational time it will require, with this relationship expected to be linear.

The second hyperparameter is the ***particle radius***. While this parameter is not directly related to the physical properties of the fluid being simulated, it significantly affects the numerical stability and resolution of the underlying SPH algorithm, and plays a crucial role in the physical fidelity of the simulation results. As the ***particle radius*** decreases, more particles are needed to represent the fluid, making the simulation more computationally expensive. Unlike the ***period of time***, this relationship **is not linear**; the number of particles required to represent the same fluid's volume increases in `O(n³)` as the particle radius decreases.

When running simulations, balancing speed and cost is critical to achieving efficient workflows. In this step, we focus on benchmarking various computational resources to identify the best machine type for our simulations. To ensure consistency, we fixed the parameters related to the ***period of time*** and ***particle radius*** (i.e., ```"stopAt": 4```, ```"timeStepSize": 0.01```, and ```"particleRadius": 0.01```, as described [here](https://tutorials.inductiva.ai/generating-synthetic-data/synthetic-data-generation-2.html#preparing-the-configuration-files)). These settings allowed us to evaluate performance across different machine types under the same conditions."

Reply:
So good!
I think we are done.